manticore/core/smtlib/solver.py

```diff
@@ -490,7 +538,7 @@ def get_all_values(self, constraints, expression, maxcnt=None, silent=False):
                     raise SolverError("Timeout")
             return result
 
-    def optimize(self, constraints: ConstraintSet, x: BitVec, goal: str, max_iter=10000):
+    def _optimize_fancy(self, constraints: ConstraintSet, x: BitVec, goal: str, max_iter=10000):
```

Comment:
Do we need to move this to a private method? It looks like it's only used internally, but there could be external scripts that depend on `optimize`. If it's solver-specific, perhaps we could make `optimize` an abstract method and implement it on the subclasses?

Reply:
Some solvers support for the `optimize` keyword. It is configured via super().__init__
https://github.com/trailofbits/manticore/pull/1714/files#diff-a32b8a365208c1e023fa207b83c84067R302-R305

Tried to have some configurable code rather than a few classes.